pyccel/decorators.py

```diff
@@ -10,66 +10,135 @@
 
 __all__ = (
     'allow_negative_index',
-    'bypass',
     'elemental',
     'inline',
     'private',
     'pure',
     'stack_array',
-    'sympy',
     'types',
 )
 
 
-def sympy(f):
-    return f
+def pure(f):
+    """
+    Add the pure keyword to the generated Fortran function.
 
-def bypass(f):
-    return f
+    Pure functions in Fortran are free from side effects. In other words, they
+    can't modify global objects or the arguments of the function, and they must
+    always produce the same output for the same inputs. This enables compiler
+    optimisations like function reordering or parallelisation.
 
-def pure(f):
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
 def private(f):
+    """
+    Indicate that a function shouldn't be exposed in the Python interface.
+
+    Indicate that a function shouldn't be exposed in the Python interface. It
+    is translated and can be called by other functions in the module but will
+    not be wrapped.
+
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
 def elemental(f):
+    """
+    Indicate that the function can also be used element-wise.
+
+    This decorator indicates that the function can be applied element-wise to
+    array arguments. The function operates on scalar inputs and the wrapper
+    will be provided for the annotatied arguments, but in the low-level code
```

Comment:
Typo: annotatied -> annotated
```suggestion
    will be provided for the annotated arguments, but in the low-level code
```